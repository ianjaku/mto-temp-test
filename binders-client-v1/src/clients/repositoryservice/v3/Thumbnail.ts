import { IRenderUrlOptions } from "./BinderVisual";
import { IThumbnail } from "./contract";
import { buildTokenUrl } from "../../authorizationservice/v1/helpers";

export default class Thumbnail implements IThumbnail {
    medium: string;
    thumbnail?: string;
    fitBehaviour: string;
    bgColor: string;
    rotation?: string;
    urlToken?: string;

    buildRenderUrl(options: IRenderUrlOptions = {}): string {
        if (options.url) {
            return buildTokenUrl(options.url, this.urlToken);
        }
        const formatNames = options.requestedFormatNames || ["medium"];
        let foundFormatName = formatNames.find(requestedFormatName => this[requestedFormatName]);
        let url = foundFormatName && this[foundFormatName];

        if (!url && this["formatUrls"]) {
            for (const requestedFormatName of formatNames) {
                const format = this["formatUrls"].find(f => f.name && f.name.toLowerCase() === requestedFormatName.toLocaleLowerCase());
                url = format?.url;
                if (url) {
                    break;
                }
            }
        }
        if (!url && this["urls"]) {
            foundFormatName = formatNames.find(requestedFormatName => this["urls"][requestedFormatName]);
            url = foundFormatName && this["urls"][foundFormatName];
        }
        if (!url) {
            // fallback to original (eg for svg's which are not transcoded to other formats)
            const origFormat = this["formatUrls"]?.find(f => f.name && f.name.toLowerCase() === "original");
            url = origFormat?.url || this["urls"]?.original || this.medium;
            if (!url) {
                // eslint-disable-next-line no-console
                console.warn("Could not find suitable render url for Thumbnail", this, "requestedFormatNames", options.requestedFormatNames, "falling back to original");
            }
        }
        if (!url) {
            // eslint-disable-next-line no-console
            console.error("Could not find suitable render url for Thumbnail", this, "requestedFormatNames", options.requestedFormatNames);
        }
        return buildTokenUrl(url, this.urlToken);
    }
}

export type WithOptionalThumbnail = { thumbnail?: IThumbnail };
export type WithThumbnail = { thumbnail?: Thumbnail };

export const withParsedThumbnail = <I>(
    item: I & WithOptionalThumbnail
): I & WithThumbnail => {
    const { thumbnail } = item;
    if (thumbnail == null) return {...item, thumbnail: null};
    const updatedThumbnail = Object.assign(Object.create(Thumbnail.prototype), thumbnail);
    return {
        // eslint-disable-next-line @typescript-eslint/ban-types
        ...(item as Object),
        thumbnail: updatedThumbnail,
    } as I & { thumbnail: Thumbnail };
}
