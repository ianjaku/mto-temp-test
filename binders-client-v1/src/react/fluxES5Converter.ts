/* eslint-disable @typescript-eslint/no-explicit-any */
// eslint-disable-next-line @typescript-eslint/explicit-module-boundary-types
export function fixES5FluxContainer<T extends new (...args: any[]) => any>(
    containerClass: T
): T {
    const tmp = containerClass;
    const newClass = function(...args: any[]) {
        return new tmp(...args);
    } as unknown as T;
    newClass.prototype = tmp.prototype;
    (newClass as any).getStores = (tmp as any).getStores;
    (newClass as any).calculateState = (tmp as any).calculateState;
    return newClass;
}
