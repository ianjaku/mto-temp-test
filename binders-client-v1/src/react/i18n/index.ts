import i18n from "../../i18n";
export { useTranslation, withTranslation } from "react-i18next";
export { type TFunction } from "../../i18n";
export default i18n;
