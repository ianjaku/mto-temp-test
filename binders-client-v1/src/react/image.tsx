/* eslint-disable @typescript-eslint/explicit-module-boundary-types */
import * as React from "react";
import { pick, replace } from "ramda";
import { isVideoURL } from "../clients/imageservice/v1/visuals";

export type Unit = "%" | "vw";

// eslint-disable-next-line @typescript-eslint/explicit-module-boundary-types
export function getFitCropStyleObject(fitBehaviour: string, aspectRatio: number, unit: Unit = "%") {
    const isLandscape = aspectRatio > 1;
    const addUnit = (amount: number) => `${amount}${unit}`;
    if (fitBehaviour === "fit") {
        if (isLandscape) {
            return {
                width: addUnit(100)
            };
        } else {
            const style = {
                height: addUnit(100),
                width: addUnit(Math.round(100 * aspectRatio))
            };
            return style;
        }
    } else {
        if (isLandscape) {
            return {
                left: addUnit(50),
                top: addUnit(50),
                height: addUnit(100),
                width: "auto",
                transform: `translate(-${addUnit(50)},-${addUnit(50)})`,
                position: "absolute"
            };
        } else {
            return {
                width: addUnit(100)
            };
        }
    }
}


export interface ImageProps {
    // eslint-disable-next-line @typescript-eslint/no-explicit-any
    image?: any;
    isOver?: boolean;
    dirtyDownSize?: boolean;
    className: string;
}

export interface ImageState {
    bgColor: string;
    width?: number;
    height?: number;
}

export default class Image extends React.Component<ImageProps, ImageState> {

    constructor(props) {
        super(props);
        this.state = {
            bgColor: props.image ? props.image.bgColor : "",
        };
    }

    handleImageLoaded(e) {
        this.setState({
            bgColor: this.props.image.bgColor,
            width: e.target.naturalWidth,
            height: e.target.naturalHeight
        });
    }

    dirtyDownsizeImageImg(img) {
        let url;
        if (img.url) {
            url = img.url;
        } else {
            url = img;
        }
        let replaceTarget = "/medium";
        if (isVideoURL(url)) {
            replaceTarget = "/video_screenshot";
        }
        return replace(/\/original/, replaceTarget, url);
    }

    getImageSrc() {
        if (this.props.dirtyDownSize) {
            return this.dirtyDownsizeImageImg(this.props.image);
        }
        if (this.props.image.buildRenderUrl) {
            return this.props.image.buildRenderUrl();
        }
        if (this.props.image.medium) {
            return this.props.image.medium;
        }
        return this.props.image;
    }

    getImageProps() {
        const { rotation }  = this.props.image;
        // eslint-disable-next-line @typescript-eslint/no-explicit-any
        const imageProps: any = {
            ...pick(["style", "alt"], this.props),
            className: this.props.isOver ? "hovering" : "",
            src: this.getImageSrc(),
            onLoad: this.handleImageLoaded.bind(this)
        };
        let rotationStyle = {};
        imageProps.style = Object.assign(imageProps.style ? imageProps.style : {}, this.getFitCropStyling());
        if(rotation) {
            const rotate = `rotate(${rotation}deg)`;
            rotationStyle = {transform:  (imageProps.style.transform) ? `${imageProps.style.transform} ${rotate}` : rotate};
            imageProps.style = {...imageProps.style, ...rotationStyle};
        }
        return imageProps;
    }

    getFitCropStyling() {
        return getFitCropStyleObject(this.props.image.fitBehaviour, this.getAspectRatio());
    }

    getAspectRatio() {
        return this.state.height ?
            this.state.width / this.state.height :
            1;
    }

    render() {
        const isTransparent = this.props.image.bgColor === "transparent";
        const wrapperStyle = { backgroundColor: isTransparent ? "inherit" : `#${this.props.image.bgColor}` } as React.CSSProperties;

        return (
            <div className={`${this.props.className ? this.props.className : ""} image-wrapper ${(isTransparent ? " transparent" : "")}`} style={wrapperStyle}>
                <div className="image-container">
                    <img {...this.getImageProps() } />
                </div>
            </div>
        );
    }

}
