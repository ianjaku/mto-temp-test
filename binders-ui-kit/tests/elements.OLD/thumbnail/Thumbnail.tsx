import * as React from "react";
import Thumbnail, { FitBehaviour } from "../../../src/elements/thumbnail";
import { create } from "react-test-renderer";
import { mount } from "enzyme";
import wrapWithTheme from "../../themeHelper";

const IMG_LANDSCAPE = "https://dummyimage.com/800x600";
const IMG_PORTRAIT = "https://dummyimage.com/600x800";

const enum Orientation {
    LANDSCAPE,
    PORTRAIT,
}

let selected = false;
let deleted = false;

const selectThumbnail = () => {
    selected = true;
};

const thumbnailOnDelete = () => {
    deleted = true;
};

const createThumbnail = (fitBehaviour: FitBehaviour, orientation: Orientation) => wrapWithTheme(
    <Thumbnail
        fitBehaviour={fitBehaviour}
        bgColor="#eeeeee"
        src={orientation === Orientation.LANDSCAPE ? IMG_LANDSCAPE : IMG_PORTRAIT} />,
);
const createInteractiveThumbnail = () => wrapWithTheme(
    <Thumbnail
        src={IMG_LANDSCAPE}
        onClick={selectThumbnail}
        isSelectable={true}
        isDeletable={true}
        onDelete={thumbnailOnDelete} />,
);

// snapshots
test("Fit landscape thumbnail (snapshot)", () => {
    const imgFitLandscape = create(createThumbnail(FitBehaviour.FIT, Orientation.LANDSCAPE));
    const serialized = imgFitLandscape.toJSON();
    expect(serialized).toMatchSnapshot();
});
test("Fit portrait thumbnail (snapshot)", () => {
    const imgFitPortrait = create(createThumbnail(FitBehaviour.FIT, Orientation.LANDSCAPE));
    const serialized = imgFitPortrait.toJSON();
    expect(serialized).toMatchSnapshot();
});
test("Crop landscape thumbnail (snapshot)", () => {
    const imgCropLandscape = create(createThumbnail(FitBehaviour.CROP, Orientation.PORTRAIT));
    const serialized = imgCropLandscape.toJSON();
    expect(serialized).toMatchSnapshot();
});
test("Crop portrait thumbnail (snapshot)", () => {
    const imgCropPortrait = create(createThumbnail(FitBehaviour.CROP, Orientation.PORTRAIT));
    const serialized = imgCropPortrait.toJSON();
    expect(serialized).toMatchSnapshot();
});

// enzyme
test("Select thumbnail (mount: enzyme+jsdom)", () => {
    const imgMount = mount(createInteractiveThumbnail());
    expect(selected).toEqual(false);
    imgMount
        .find(".thumbnail-wrapper")
        .first()
        .simulate("click");
    expect(selected).toEqual(true);
});

test("Delete thumbnail (mount: enzyme+jsdom)", () => {
    const imgMount = mount(createInteractiveThumbnail());
    expect(deleted).toEqual(false);
    imgMount
        .find(".thumbnail-wrapper")
        .simulate("mouseEnter");

    imgMount
        .find(".thumbnail-outer-wrapper .material-icons")
        .first()
        .simulate("click");

    expect(deleted).toEqual(true);
});
