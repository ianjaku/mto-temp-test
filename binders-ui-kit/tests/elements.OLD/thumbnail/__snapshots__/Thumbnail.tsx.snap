// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Crop landscape thumbnail (snapshot) 1`] = `
<div
  className="thumbnail-outer-wrapper"
  style={
    Object {
      "backgroundColor": "##eeeeee",
      "height": undefined,
      "width": undefined,
    }
  }
>
  <div
    className="thumbnail-wrapper crop landscape"
    onClick={undefined}
    onDoubleClick={undefined}
    onMouseDownCapture={undefined}
    onMouseEnter={[Function]}
    onMouseLeave={[Function]}
    style={
      Object {
        "backgroundColor": "##eeeeee",
        "height": undefined,
        "width": undefined,
      }
    }
  >
    <img
      className="thumbnail crop landscape"
      onLoad={[Function]}
      src="https://dummyimage.com/600x800"
      style={
        Object {
          "height": undefined,
          "left": "50%",
          "position": "absolute",
          "top": "50%",
          "transform": "translate(-50%, -50%)",
          "width": "auto",
        }
      }
      title="https://dummyimage.com/600x800"
    />
  </div>
</div>
`;

exports[`Crop portrait thumbnail (snapshot) 1`] = `
<div
  className="thumbnail-outer-wrapper"
  style={
    Object {
      "backgroundColor": "##eeeeee",
      "height": undefined,
      "width": undefined,
    }
  }
>
  <div
    className="thumbnail-wrapper crop landscape"
    onClick={undefined}
    onDoubleClick={undefined}
    onMouseDownCapture={undefined}
    onMouseEnter={[Function]}
    onMouseLeave={[Function]}
    style={
      Object {
        "backgroundColor": "##eeeeee",
        "height": undefined,
        "width": undefined,
      }
    }
  >
    <img
      className="thumbnail crop landscape"
      onLoad={[Function]}
      src="https://dummyimage.com/600x800"
      style={
        Object {
          "height": undefined,
          "left": "50%",
          "position": "absolute",
          "top": "50%",
          "transform": "translate(-50%, -50%)",
          "width": "auto",
        }
      }
      title="https://dummyimage.com/600x800"
    />
  </div>
</div>
`;

exports[`Fit landscape thumbnail (snapshot) 1`] = `
<div
  className="thumbnail-outer-wrapper"
  style={
    Object {
      "backgroundColor": "##eeeeee",
      "height": undefined,
      "width": undefined,
    }
  }
>
  <div
    className="thumbnail-wrapper fit landscape"
    onClick={undefined}
    onDoubleClick={undefined}
    onMouseDownCapture={undefined}
    onMouseEnter={[Function]}
    onMouseLeave={[Function]}
    style={
      Object {
        "backgroundColor": "##eeeeee",
        "height": undefined,
        "width": undefined,
      }
    }
  >
    <img
      className="thumbnail fit landscape"
      onLoad={[Function]}
      src="https://dummyimage.com/800x600"
      style={
        Object {
          "height": "auto",
          "width": undefined,
        }
      }
      title="https://dummyimage.com/800x600"
    />
  </div>
</div>
`;

exports[`Fit portrait thumbnail (snapshot) 1`] = `
<div
  className="thumbnail-outer-wrapper"
  style={
    Object {
      "backgroundColor": "##eeeeee",
      "height": undefined,
      "width": undefined,
    }
  }
>
  <div
    className="thumbnail-wrapper fit landscape"
    onClick={undefined}
    onDoubleClick={undefined}
    onMouseDownCapture={undefined}
    onMouseEnter={[Function]}
    onMouseLeave={[Function]}
    style={
      Object {
        "backgroundColor": "##eeeeee",
        "height": undefined,
        "width": undefined,
      }
    }
  >
    <img
      className="thumbnail fit landscape"
      onLoad={[Function]}
      src="https://dummyimage.com/800x600"
      style={
        Object {
          "height": "auto",
          "width": undefined,
        }
      }
      title="https://dummyimage.com/800x600"
    />
  </div>
</div>
`;
